### PR TITLE
Disallow missing field initializers

### PR DESCRIPTION
**Pull request by tamird**
This warning is enabled in Fuchsia, and appeasing it is quite trivial.

@haberman @fowles

### CODE REVIEW

File: src/google/protobuf/arena.cc [hunk 1]
```diff
@@ -115,7 +115,11 @@ SerialArena::SerialArena(Block* b, void* owner) : space_allocated_(b->size) {
 SerialArena* SerialArena::New(Memory mem, void* owner) {
   GOOGLE_DCHECK_LE(kBlockHeaderSize + ThreadSafeArena::kSerialArenaSize, mem.size);
 
-  auto b = new (mem.ptr) Block{nullptr, mem.size};
+  auto b = new (mem.ptr) Block{
+      .next = nullptr,
+      .size = mem.size,
+      .start = nullptr,
+  };
```
Comment (fowles) [inline src/google/protobuf/arena.cc:122]: Designated initializers are a c++20 extension. We support c++11

File: src/google/protobuf/compiler/cpp/cpp_helpers.cc [hunk 2]
```diff
@@ -1130,7 +1130,7 @@ bool IsImplicitWeakField(const FieldDescriptor* field, const Options& options,
 
 MessageAnalysis MessageSCCAnalyzer::GetSCCAnalysis(const SCC* scc) {
   if (analysis_cache_.count(scc)) return analysis_cache_[scc];
-  MessageAnalysis result{};
+  MessageAnalysis result = {};
```
Comment (fowles) [inline src/google/protobuf/compiler/cpp/cpp_helpers.cc:1133]: This one should be unneeded

File: src/google/protobuf/arena.cc (hunk 1 above)
Comment (tamird) [reply src/google/protobuf/arena.cc:122]: Done.

File: src/google/protobuf/compiler/cpp/cpp_helpers.cc (hunk 2 above)
Comment (tamird) [reply src/google/protobuf/compiler/cpp/cpp_helpers.cc]: You're right - not sure what I was seeing, but it doesn't seem to be needed.

File: src/google/protobuf/compiler/cpp/cpp_helpers.cc [hunk 3]
```diff
@@ -1130,7 +1130,7 @@ bool IsImplicitWeakField(const FieldDescriptor* field, const Options& options,
 
 MessageAnalysis MessageSCCAnalyzer::GetSCCAnalysis(const SCC* scc) {
   if (analysis_cache_.count(scc)) return analysis_cache_[scc];
-  MessageAnalysis result{};
+  MessageAnalysis result{false, false, false, false, false};
```
Comment (fowles) [inline src/google/protobuf/compiler/cpp/cpp_helpers.cc:1133]: blech, I kinda prefer adding defaults to this struct instead

Comment (tamird) [reply src/google/protobuf/compiler/cpp/cpp_helpers.cc]: Done.

**Review by fowles — APPROVED** [review]

**Pull request merged**